lib/bin/supllib/include/supl_session.h

```diff
@@ -0,0 +1,317 @@
+ * DISCARDED BY ANY JURISDICTION, NORDIC SEMICONDUCTOR ASA, ITS CONTRIBUTORS
+ * AND THIRD PARTY LICENSORS’ LIABILITY SHALL IN NO CIRCUMSTANCE EXCEED AN
+ * AGGREGATE OF 50 USD FOR ALL CLAIMS.
+ */
+
+#ifndef SUPL_SUPL_SESSION_H_
+#define SUPL_SUPL_SESSION_H_
+
+#include <stdbool.h>
+
+#include <nrf_socket.h>
+
+#define LIBSUPL_SEC_BUF_SIZE                (4096)
+#define LIBSUPL_PRI_BUF_SIZE                (LIBSUPL_SEC_BUF_SIZE * 3)
+
+#define LIBSUPL_LOG_LEVEL_NONE              (0)
+#define LIBSUPL_LOG_LEVEL_ERR               (1)
+#define LIBSUPL_LOG_LEVEL_WRN               (2)
+#define LIBSUPL_LOG_LEVEL_INF               (3)
+#define LIBSUPL_LOG_LEVEL_DBG               (4)
+
+/* SUPL session result codes */
+#define LIBSUPL_PARAM_ERR                   (-17)
+#define LIBSUPL_ABORT                       (-16)
+#define LIBSUPL_DECODER_ERR                 (-15)
+#define LIBSUPL_UNSPECIFIED_ERR             (-14)
+#define LIBSUPL_TIMEOUT                     (-13)
+#define LIBSUPL_INJECTION_SOCKET_ERR        (-12)
+#define LIBSUPL_INJECTION_INVALID_DATA      (-11)
+#define LIBSUPL_INJECTION_NO_DATA           (-10)
+#define LIBSUPL_SESSION_INVALID_ERR         (-9)
+#define LIBSUPL_SESSION_MISSING_ERR         (-8)
+#define LIBSUPL_SESSION_SLP_MISMATCH_ERR    (-7)
+#define LIBSUPL_SESSION_SET_MISMATCH_ERR    (-6)
+#define LIBSUPL_SESSION_EXIST_ERR           (-5)
+#define LIBSUPL_INVALID_MSG_ERR             (-4)
+#define LIBSUPL_SOCKET_ERR                  (-3)
+#define LIBSUPL_LPP_DECODE_ERR              (-2)
+#define LIBSUPL_ULP_DECODE_ERR              (-1)
+#define LIBSUPL_SUCCESS                     (0)
+
+/* Device ID
+ * Formats how device_id array SHALL be filled for different choices:
+ *
+ * IMSI
+ * Example 244121303566399
+ * device_id[0] = 0x24;
+ * device_id[1] = 0x41;
+ * device_id[2] = 0x21;
+ * device_id[3] = 0x30;
+ * device_id[4] = 0x35;
+ * device_id[5] = 0x66;
+ * device_id[6] = 0x39;
+ * device_id[7] = 0x90;
+ *
+ * IPV4
+ * Example IP 123.45.67.89
+ * device_id[0] = 123;
+ * device_id[1] = 45;
+ * device_id[2] = 67;
+ * device_id[3] = 89;
+ *
+ * IPV6
+ * Example IP 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+ * device_id[0] = 0x20;
+ * device_id[1] = 0x01;
+ * device_id[2] = 0x0d;
+ * device_id[3] = 0xb8;
+ * device_id[4] = 0x85;
+ * device_id[5] = 0xa3;
+ * device_id[6] = 0x00;
+ * device_id[7] = 0x00;
+ * device_id[8] = 0x00;
+ * device_id[9] = 0x00;
+ * device_id[10] = 0x8a;
+ * device_id[11] = 0x2e;
+ * device_id[12] = 0x03;
+ * device_id[13] = 0x70;
+ * device_id[14] = 0x73;
+ * device_id[15] = 0x34;
+ */
+#define LIBSUPL_ID_CHOICE_IMSI          (1)
+#define LIBSUPL_ID_CHOICE_IPV4          (2)
+#define LIBSUPL_ID_CHOICE_IPV6          (3)
+
+#define LIBSUPL_ID_LEN                  (16)
+
+
+struct lte_params {
+	bool status;
+	unsigned short mcc[3];
+	int mnc_length;
+	unsigned short mnc[3];
+	unsigned char tac[2];
+	unsigned char cid[4];
+	unsigned short phys_cid;
```

Comment:
Are we in control of this code?

If so, can we use u16_t here?

Reply:
yes, but I can not use the zephyr defines as the supl library is OS agnostic. I can change these to stdint.h types.

Reply:
It's OS agnostic?

But we are using char2hex for instance, why is that OK, but not u32_t ?

Reply:
char2hex is only used in the OS integration part and is not a part of the library. supl_session.h is a part of the library. To make it cleaner I have removed the copy in nrf and it will only exist in the downloaded zip file.